php/libraries/User.class.inc

```diff
@@ -508,6 +508,7 @@ class User extends UserPermissions
      */
     function updatePassword($password, $expiry = null)
     {
+        $password = htmlspecialchars_decode($password);
```

Comment:
I think this should be done before the call to updatePassword. What's encoding it? A different code path might not and would introduce a similar bug because it doesn't match the caller's expectations.

Reply:
Yeah the password shouldn't get encoded this way. We should track down the code that's calling htmlspecialchars and remove it (or replace it with base64 which would be more appropriate).